### PR TITLE
Core: Change PostgreSQL to use check constraints rather than enums #2436

### PR DESCRIPTION
**Pull request by illingwo**
Core: Change PostgreSQL to use check constraints rather than enums #2436

### CODE REVIEW

File: lib/rucio/db/sqla/migrate_repo/versions/f1b14a8c2ac1_postgres_use_check_constraints.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+revision = 'f1b14a8c2ac1'
+down_revision = 'b8caac94d7f0'
+
+
+def upgrade():
+    '''
+    Upgrade the database to this revision
+    '''
+
+    if context.get_context().dialect.name != 'postgresql':
```
Comment (mlassnig) [inline lib/rucio/db/sqla/migrate_repo/versions/f1b14a8c2ac1_postgres_use_check_constraints.py:34]: i'm a friend of negations as much as anyone else, but wouldn't an indented block with `dialect == "postgresql"` be more descriptive? :-D

Comment (illingwo) [reply lib/rucio/db/sqla/migrate_repo/versions/f1b14a8c2ac1_postgres_use_check_constraints.py:34]: You can see that it only applies to PostgreSQL right at the beginning without having to scroll to the end to look for an else clause (and originally I thought that the whole thing would be even longer than it turned out).

Comment (mlassnig) [reply lib/rucio/db/sqla/migrate_repo/versions/f1b14a8c2ac1_postgres_use_check_constraints.py:34]: it's fine with me ;-)

**Review by mlassnig — APPROVED** [review]

File: lib/rucio/db/sqla/migrate_repo/versions/f1b14a8c2ac1_postgres_use_check_constraints.py (hunk 1 above)
Comment (bari12) [reply lib/rucio/db/sqla/migrate_repo/versions/f1b14a8c2ac1_postgres_use_check_constraints.py:34]: Just to confirm, for every migrate script after this one, we can simply rely on the `create_constraint/drop_constraint` syntax of alembic, correct?

Comment (illingwo) [reply lib/rucio/db/sqla/migrate_repo/versions/f1b14a8c2ac1_postgres_use_check_constraints.py:34]: I believe so. This migration script should make the constraints consistent, so there will always be something for `drop_constraint` to drop, and the type will be correct for `create_check_constraint`.

Comment (illingwo) [reply lib/rucio/db/sqla/migrate_repo/versions/f1b14a8c2ac1_postgres_use_check_constraints.py:34]: As a quick test I took the migrate script from #1864, changed the postgresql upgrade to use the  same code block as Oracle with just `drop_constraint/create_check_constraint`, and it applies cleanly.

**Pull request merged**